crates/bevy_ecs/src/world/mod.rs

```diff
@@ -96,13 +97,52 @@ use unsafe_world_cell::{UnsafeEntityCell, UnsafeWorldCell};
 ///     commands.queue(AddToCounter(42));
 /// }
 /// ```
-pub trait Command: Send + 'static {
+pub trait Command<Marker = ()>: Send + 'static {
     /// Applies this command, causing it to mutate the provided `world`.
     ///
     /// This method is used to define what a command "does" when it is ultimately applied.
     /// Because this method takes `self`, you can store data or settings on the type that implements this trait.
     /// This data is set by the system or other source of the command, and then ultimately read in this method.
-    fn apply(self, world: &mut World);
+    fn apply(self, world: &mut World) -> Result;
+
+    /// Returns a new [`Command`] that, when applied, will apply the original command
+    /// and handle any resulting errors according to the provided [`CommandErrorMode`].
+    fn with_error_handling(
+        self,
+        error_mode: CommandErrorMode,
+    ) -> impl FnOnce(&mut World) + Send + 'static
+    where
+        Self: Sized,
+    {
+        move |world: &mut World| {
+            if let Err(error) = self.apply(world) {
+                match error_mode {
+                    CommandErrorMode::Silent => (),
+                    CommandErrorMode::Warn => warn!("Command returned an error: {error}"),
+                    CommandErrorMode::Error => error!("Command returned an error: {error}"),
+                    CommandErrorMode::Panic => panic!("Command returned an error: {error}"),
+                    CommandErrorMode::Override(function) => function(),
+                }
+            }
+        }
+    }
+}
+
+/// How a [`Command`] should respond upon encountering an [`Error`](crate::result::Error).
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub enum CommandErrorMode {
+    /// The command will fail silently.
+    Silent,
+    /// The command will send a warning.
+    Warn,
+    /// The command will send an error.
+    Error,
+    /// The command will start a panic.
+    Panic,
+    /// The command will call a custom function.
+    ///
+    /// `fn()` can be a closure if it doesn't capture its environment.
+    Override(fn()),
```

Comment:
In order for this to be useful, we need to pass in both the `Result` (or at least the error arm) and `&mut World` to actually respond. Without any data access this is only really viable for custom logging.